packages/sdk/src/lib/dataProtectorSharing/setProtectedDataForSale.ts

```diff
@@ -38,116 +39,68 @@ export const setProtectedDataForSale = async ({
     .label('priceInNRLC')
     .validateSync(priceInNRLC);
 
-  const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
-
-  const protectedData = await checkAndGetProtectedData({
-    graphQLClient,
-    protectedDataAddress: vProtectedDataAddress,
-    userAddress,
-  });
-
+  let userAddress = await iexec.wallet.getAddress();
+  userAddress = userAddress.toLowerCase();
   const sharingContract = await getSharingContract(
     iexec,
     sharingContractAddress
   );
-  const tx = await sharingContract.setProtectedDataForSale(
-    protectedData.collection.id,
-    protectedData.id,
-    vPriceInNRLC
-  );
-  await tx.wait();
-
-  await waitForSubgraphIndexing();
-
-  return {
-    success: true,
-    txHash: tx.hash,
-  };
-};
 
-async function checkAndGetProtectedData({
-  graphQLClient,
-  protectedDataAddress,
-  userAddress,
-}: {
-  graphQLClient: GraphQLClient;
-  protectedDataAddress: Address;
-  userAddress: Address;
-}) {
-  const { protectedData } = await getProtectedDataById({
-    graphQLClient,
-    protectedDataAddress,
+  const collectionTokenId = await getCollectionForProtectedData({
```

Comment:
`collectionTokenId` can be undefined here, we need to check for that.

There was an explicit check in previous code: 'This protected data does not exist in the subgraph.'